### PR TITLE
Add Factory Method to SynonymMap to Create from File

### LINKED ISSUE #20329: Add a Convenience Method to Create a SynonymMap from a File/Stream
Add a convenience method, or constructor, which allows creation of a synonym map from a file or stream.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #20329 

This PR adds a factory method to `SynonymMap` to enable creating it from a given file.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/util/Utility.java [hunk 1]
```diff
@@ -169,6 +173,14 @@ public static synchronized String formatCoordinate(double coordinate) {
         return COORDINATE_FORMATTER.format(coordinate);
     }
 
+    public static String readSynonymsFromFile(Path filePath) {
+        try {
+            return new String(Files.readAllBytes(filePath), StandardCharsets.UTF_8);
+        } catch (IOException ex) {
+            throw new ClientLogger(Utility.class).logExceptionAsError(new UncheckedIOException(ex));
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/util/Utility.java:180]: nit: is that better to have a static LOGGER? I know it is only used once but just want to discuss it.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SynonymMap.java [hunk 2]
```diff
@@ -159,4 +159,17 @@ public SynonymMap setETag(String eTag) {
...
         return this;
     }
+
+    /**
+     * Creates a new instance of SynonymMap with synonyms read from the passed file.
+     *
+     * @param name The name of the synonym map.
+     * @param filePath The path to the file where the formatted synonyms are read.
+     * @return A SynonymMap.
+     * @throws java.io.UncheckedIOException If reading {@code file} fails.
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SynonymMap.java:169]: {@code file} or {@code filePath}?

**Review by mssfang — APPROVED** [review]

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SynonymMap.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SynonymMap.java:169]: Should be filePath, good catch

**Pull request merged**